packages/amplify-environment-parameters/API.md

```diff
@@ -44,6 +46,25 @@ export class ResourceParameterManager {
 // @public (undocumented)
 export const saveAll: () => Promise<void>;
 
+// @public (undocumented)
+export class SSMClientWrapper {
```

Comment:
if the intent is to store environmental secrets then wouldn't it be better if
SSMClientWrapper becomes internal impl detail and we create `SecretEnvironmentParameterManager implements IEnvironmentParameterManager` that uses SSMClientWrapper under the hood?
Or IEnvironmentParameterManager accepts parameter in its apis that flags env parameter as secret ?

Reply:
For better or worse, the two interfaces aren't really compatible. The intent here also isn't to store secrets, the intent is we need to interact with parameter store for non secret values in the environment parameter manager and want to reuse the logic in SSMWrapper in both the env param manager and the function secrets code. But we also don't want to change too much about the current function env vars / secrets implementation.

Reply:
Ah shoot, I had a response written that I never hit "comment" on. I agree that the original naming does leak the implementation details and doesn't make for the best package export. What about changing the name to something like `CloudEnvironmentParameterManager`?

Reply:
@edwardfoyle if that is the case, then shouldn't `SSMWrapper` go to cli core ?

Reply:
Adding code that interacts with aws-sdk calls, even though it's an injected dependency, increases the scope of cli-core which I would prefer to avoid.

Thoughts on renaming the class?